spec/lib/task-spec.js

```diff
@@ -242,6 +242,39 @@ describe("Task", function () {
             expect(task.options.nested3).to.equal(definition.options.sourceValue);
         });
 
+        it("should render iteration templates", function() {
+            definition.options = {
+                testList: [
+                    {
+                        name: 'item 1'
+                    },
+                    {
+                        name: 'item 2'
+                    },
+                    {
+                        name: 'item 3'
+                    }
+                ],
+                testVal: '{{#options.testList}}{{ name }}.{{/options.testList}}'
+            };
+            var task = Task.create(definition, {}, {});
+            var tempList = _.transform(definition.options.testList, function (result, n) {
+                result.push(n.name);
+            });
+            expect(task.options.testVal).to.equal(tempList.join('.') + '.');
+        });
+
+        it("should render condition templates", function() {
+            definition.options = {
+                testSrc1: 'Test source 1 exist',
+                testVal1: '{{#options.testSrc1}}{{ options.testSrc1 }}{{/options.testSrc1}}',
+                testVal2: '{{#options.testSrc2}}{{ options.testSrc2 }}{{/options.testSrc2}}'
+            };
+            var task = Task.create(definition, {}, {});
+            expect(task.options.testVal1).to.equal(definition.options.testSrc1);
+            expect(task.options.testVal2).to.equal('');
```

Comment:
I think we can keep the way which normal hogan rendering. Since the certain value was put in the condtional section {{#condition}}  {{/condtion}}. It's reasonable to skip anything inside the condition if the condition doesn't meet. 
While, if we just put an not existing value {{value}} to template, hogan would also report error during render time.